src/common/net/asio_http_cli.h

```diff
@@ -550,7 +550,7 @@ class AsioHttpClientPool
             co_return std::shared_ptr<AsioHttpClient>();
           }
 
-          auto client_key = client_options.host + client_options.service;
+          auto client_key = client_options.host + ":" + client_options.service;  // 确保 client_key 的唯一性
```

Comment:
This consideration is practical; such cases as 127.0.0.12:3 and 127.0.0.1:23 could indeed occur. However, simply adding a colon might not be a good solution, as IPv6 addresses already contain many colons, and there might still be some unknown issues.
We could consider writing a struct with a hash method. You can refer to

https://github.com/AimRT/AimRT/blob/69b83817cd2a5cbdcb4627da08c006769ec4ac99/src/plugins/record_playback_plugin/topic_meta_key.h#L10-L24